src/main/java/net/dv8tion/jda/core/entities/impl/InviteImpl.java

```diff
@@ -117,6 +121,9 @@ protected void handleResponse(final Response response, final Request<Void> reque
         if (this.expanded)
             return new RestAction.EmptyRestAction<>(getJDA(), this);
 
+        if (this.type != Invite.InviteType.GUILD)
+            throw new UnsupportedOperationException("Only guild invites can be expanded");
```

Comment:
I can't see where this exception is documented. I would also prefer using `IllegalStateException` since the state is what makes the difference here, not the operation.

Reply:
[`Invite#expand()`](https://github.com/DV8FromTheWorld/JDA/pull/709/files#diff-ed6e83754ad768ec7d551332e8cbed3cR102):
>As bots can't be in groups this is only available for guild invites and will throw an _UnsupportedOperationException_ for other types.

>@throws java.lang.UnsupportedOperationException
>If this is a group invite

Also you can get more info about a group invite if you're in the group (`created_at`, `max_age`) so it technically is an unsupported operation. Have no problems  it to an`IllegalStateException` though.

Reply:
Oh looks like I misread `expand` as `resolve`, my bad. But yeah I would rather use `IllegalStateException` for this.